homeassistant/components/subaru/lock.py

```diff
@@ -0,0 +1,52 @@
+"""Support for Subaru door locks."""
+from homeassistant.components.lock import DOMAIN as LOCK_DOMAIN, LockEntity
+from homeassistant.const import SERVICE_LOCK, SERVICE_UNLOCK
+
+from . import DOMAIN as SUBARU_DOMAIN
+from .const import (
+    ENTRY_CONTROLLER,
+    ENTRY_COORDINATOR,
+    ENTRY_VEHICLES,
+    VEHICLE_HAS_REMOTE_SERVICE,
+)
+from .entity import SubaruEntity
+from .remote_service import async_call_remote_service
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Subaru locks by config_entry."""
+    controller = hass.data[SUBARU_DOMAIN][config_entry.entry_id][ENTRY_CONTROLLER]
+    coordinator = hass.data[SUBARU_DOMAIN][config_entry.entry_id][ENTRY_COORDINATOR]
+    vehicle_info = hass.data[SUBARU_DOMAIN][config_entry.entry_id][ENTRY_VEHICLES]
+    entities = []
+    for vin in vehicle_info:
+        if vehicle_info[vin][VEHICLE_HAS_REMOTE_SERVICE]:
+            entities.append(SubaruLock(vehicle_info[vin], coordinator, controller))
```

Comment:
suggestion:
this could be shortened slightly to:
```
for vehicle in vehicle_info.values():
    if vehicle[VEHICLE_HAS_REMOTE_SERVICE]:
        entities.append(SubaruLock(vehicle, coordinator, controller))
```